watertap/unit_models/ion_exchange_0D.py

```diff
@@ -141,85 +246,103 @@ class IonExchangeODData(InitializationMixin, UnitModelBlockData):
+        self.process_flow = ControlVolume0DBlock(
+            dynamic=False,
+            has_holdup=False,
+            property_package=self.config.property_package,
+            property_package_args=self.config.property_package_args,
         )
 
-        self.underdrain_h = Param(
-            initialize=0.5, units=pyunits.m, doc="Underdrain height"  # Perry's
+        self.process_flow.add_state_blocks(has_phase_equilibrium=False)
+        self.process_flow.add_material_balances(
+            balance_type=self.config.material_balance_type, has_mass_transfer=True
         )
-
-        self.distributor_h = Param(
-            initialize=0.5, units=pyunits.m, doc="Distributor height"  # Perry's
+        self.process_flow.add_energy_balances(
+            balance_type=self.config.energy_balance_type, has_enthalpy_transfer=False
+        )
+        self.process_flow.add_isothermal_assumption()
+        self.process_flow.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=False,
         )
 
-        self.p_drop_psi_to_m = Param(
-            initialize=0.70325,
-            units=(pyunits.m / pyunits.psi),
-            doc="Conversion for pressure drop in psi to m",
+        prop_in = self.process_flow.properties_in[0]
+
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = False
+
+        self.regeneration_stream = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of regeneration stream",
+            **tmp_dict,
         )
 
-        # Liquid holdup correlation
-        # Eq. 4.101 in Inamuddin/Luqman
+        regen = self.regeneration_stream[0]
 
-        self.holdup_A = Param(
-            initialize=21,
-            units=pyunits.dimensionless,
-            doc="Holdup equation A parameter",
+        self.add_inlet_port(name="inlet", block=self.process_flow)
+        self.add_outlet_port(name="outlet", block=self.process_flow)
+        self.add_outlet_port(name="regen", block=self.regeneration_stream)
+
+        # ==========PARAMETERS==========
+
+        self.underdrain_h = Param(
+            initialize=0.5,
+            mutable=True,
+            units=pyunits.m,
+            doc="Underdrain height",  # Perry's
         )
 
-        self.holdup_B = Param(
-            initialize=99.72,
-            units=pyunits.dimensionless,
-            doc="Holdup equation B parameter",
+        self.distributor_h = Param(
+            initialize=0.5,
+            mutable=True,
+            units=pyunits.m,
+            doc="Distributor height",  # Perry's
         )
 
-        self.holdup_exp = Param(
-            initialize=0.28, units=pyunits.dimensionless, doc="Holdup equation exponent"
+        self.p_drop_psi_to_m = Param(
```

Comment:
You know this works, right? 
```
    pyunits.convert(
        1*pyunits.psi/(998*pyunits.kg*pyunits.m**-3)/(9.81*pyunits.m*pyunits.second**-2),
                       to_units=pyunits.m)
```
I confirmed its ~0.703 using P=rho*g*h, which I think you just revert these calculations later. You are not actually doing a conversion, but instead using the pressure head equation, which isn't clear from this. I would say for conciseness this should be revised.

Reply:
-_- ... yes of course I knew that. 

Addressed in my last commit. Turns out you don't need density or gravity at all.